### ISSUE TITLE
Cost Of Goods Sold Account must be used with an Inventory Asset Account

### ISSUE THREAD

**Issue by talisman-innovations** [calcinai/xero-php#264]
Hi when trying to add a new item via the lib I have been constantly getting the error in the subject.

I tracked it down to adding a purchase and sale object to the item. the xml produced had a sub node 

i.e.

<Item><Code>lightspeed_22</Code><InventoryAssetAccountCode>630</InventoryAssetAccountCode><Name>Bespoke Waterloo Coaster</Name><IsSold>true</IsSold><IsPurchased>true</IsPurchased><PurchaseDetails>**<PurchaseDetail>**<UnitPrice>1.25</UnitPrice><COGSAccountCode>310</COGSAccountCode><TaxType>CAPEXINPUT2</TaxType>**</PurchaseDetail>**</PurchaseDetails><SalesDetails><SalesDetail><UnitPrice>3.333</UnitPrice><AccountCode>200</AccountCode><TaxType>CAPEXOUTPUT2</TaxType></SalesDetail></SalesDetails><IsTrackedAsInventory>true</IsTrackedAsInventory></Item>

When it fix up to remove the singular **</PurchaseDetail>** it adds ok.

I traced the issue is in the Helper file aro line 28. 

Am I using it incorrectly to add purchases and sales to products?

anyhoo here's the code I've put in to fix the issue for me

//recurse and replace.
                if(self::isAssoc($element))
                    $element = self::arrayToXML($element);
                else if (count($element)==1)
                    $element = self::arrayToXML($element[0]);
                else
                    $element = self::arrayToXML($element, self::singularize($key));

**Comment by calcinai** [comment]
Hey @talisman-innovations, it's not 100% clear what the fix is here - can you please make a diff or a PR for it?

**Comment by laurenkatherine** [comment]
Hi,
We had the same issue here and found the XML file produced was as below:
<PurchaseDetails>
        <PurchaseDetail>
            <COGSAccountCode>310</COGSAccountCode>
        </PurchaseDetail>
</PurchaseDetails>

where the singular PurchaseDetail does not appear to be needed.

In the addPurchaseDetail function of Item.php we removed the second set of square brackets from 

$this->_data['PurchaseDetails'][] = $value;

and it now seems to be working. We also did the same for addSalesDetail. 

Hopefully this fixes it, but if there was a reason it was needed and we might encounter another problem later without it, please let us know.

**Comment by laurenkatherine** [comment]
Hi,
We had the same issue here and found the XML file produced was as below:

```
<PurchaseDetails>
        <PurchaseDetail>
            <COGSAccountCode>310</COGSAccountCode>
        </PurchaseDetail>
</PurchaseDetails>
```

where the singular PurchaseDetail does not appear to be needed.

In the addPurchaseDetail function of Item.php we removed the second set of square brackets from 

`$this->_data['PurchaseDetails'][] = $value;`

and it now seems to be working. We also did the same for addSalesDetail. 

Hopefully this fixes it, but if there was a reason it was needed and we might encounter another problem later without it, please let us know.

**Comment by alexbowers** [comment]
This has been fixed in master. RElease needed. Was fixed in #308

**Comment by calcinai** [comment]
@alexbowers done!
